src/ClientHandle.cpp

```diff
@@ -1243,13 +1246,32 @@ void cClientHandle::HandleBlockDigFinished(int a_BlockX, int a_BlockY, int a_Blo
 		}
 	}
 
+	if (!m_Player->IsGameModeCreative() && !cBlockInfo::IsOneHitDig(a_OldBlock))
+	{
+		// Fix for very fast tools.
+		m_BreakProgress += m_Player->GetPlayerRelativeBlockHardness(a_OldBlock);
+		if (m_BreakProgress < FASTBREAK_PERCENTAGE)
+		{
+			LOGD("Break progress of player %s was less than expected: %f < %f\n", m_Player->GetName().c_str(), m_BreakProgress * 100, FASTBREAK_PERCENTAGE * 100);
+			// AntiFastBreak doesn't agree with the breaking. Bail out. Send the block back to the client, so that it knows:
+			m_Player->GetWorld()->SendBlockTo(a_BlockX, a_BlockY - 1, a_BlockZ, m_Player);  // Bug with doors and things like that.
+			m_Player->GetWorld()->SendBlockTo(a_BlockX, a_BlockY, a_BlockZ, m_Player);
+			m_Player->GetWorld()->SendBlockTo(a_BlockX, a_BlockY + 1, a_BlockZ, m_Player);  // Bug with doors and things like that.
```

Comment:
Actually, I have the opposite comment.  I'm 99% sure that there are *no* blocks that change neighboring blocks clientside.  Doors only do it serverside.  Doubleplants only do it serverside (and thus experience [graphical glitches](https://bugs.mojang.com/browse/MC-53439) in some cases).  Beds only do it serverside, in all cases (both exploding in the nether and breaking).  Vines only do it serverside.  Etcetera.  So you really only should need to repair the modified block and not the ones above/below/near it.